internal/git/git.go

```diff
@@ -14,44 +15,55 @@ const (
 
 type checkoutCmd struct {
 	rukpakv1alpha1.GitSource
+	User  string
+	Token string
 }
 
-func CloneCommandFor(s rukpakv1alpha1.GitSource) (string, error) {
-	cmd := &checkoutCmd{GitSource: s}
+func CloneCommandFor(s rukpakv1alpha1.GitSource, user, token string) (string, error) {
+	cmd := &checkoutCmd{GitSource: s, User: user, Token: token}
 	err := cmd.Validate()
 	if err != nil {
 		return "", err
 	}
-	return cmd.String(), nil
+	return cmd.String()
 }
 
-func (c *checkoutCmd) String() string {
+func (c *checkoutCmd) String() (string, error) {
 	var checkoutCommand string
 	var repository = c.Repository
 	var directory = c.Directory
 	var branch = c.Ref.Branch
 	var commit = c.Ref.Commit
 	var tag = c.Ref.Tag
 
+	if c.User != "" || c.Token != "" {
+		repositoryURL, err := url.Parse(repository)
+		if err != nil {
+			return "", err
+		}
+		repositoryURL.User = url.UserPassword(c.User, c.Token)
+		repository = repositoryURL.String()
```

Comment:
This will result in the plaintext credentials being in the pod spec of the git unpack pod, which IMO is a non-starter.

Is there a way to mount the credentials as a volume or set environment variables with `valueFrom` and have git read from there?

Reply:
Yes, I'll try the environment variables.